zerver/lib/actions.py

```diff
@@ -2703,6 +2703,39 @@ def check_schedule_message(
     return do_schedule_messages([send_request])[0]
 
 
+def validate_message_edit_payload(
+    message: Message,
+    stream_id: Optional[int],
+    topic_name: Optional[str],
+    propagate_mode: Optional[str],
+    content: Optional[str],
+) -> None:
+    """
+    Checks that the data sent is well-formed. Does not handle editability, permissions etc.
+    """
+    if topic_name is None and content is None and stream_id is None:
+        raise JsonableError(_("Nothing to change"))
+
+    if not message.is_stream_message():
+        if stream_id is not None:
+            raise JsonableError(_("Private messages cannot be moved to streams."))
+
```

Comment:
I changed this error message -- it was `Message must be a stream message` earlier, which I think does not explain completely why the request failed.

Reply:
Sounds good!